cpp/tests/utility/FileSystem.cpp

```diff
@@ -353,11 +353,18 @@ TEST(FileSystem, DeleteDirectory) {
     status = utility::filesystem::MakeDirectory(path);
     EXPECT_TRUE(status);
 
+    status = utility::filesystem::MakeDirectory(path + "/sub_dir");
+    EXPECT_TRUE(status);
+    FILE *file = utility::filesystem::FOpen(path + "/file_name.txt", "w");
+    fclose(file);
+
     status = utility::filesystem::DeleteDirectory(path);
     EXPECT_TRUE(status);
 
+    // std::filesystem::remove_all() won't throw error if the directory doesn't
```

Comment:
Do we want to use `utility::filesystem::DirectoryExists()` to check, and throw a warning if the directory does not exist?

Reply:
I'd like it to behave like `rm -rf`, which will be silent if the dir does not exist.